library-impl/src/main/scala/special/collection/CollsOverArrays.scala

```diff
@@ -59,7 +60,12 @@ class CollOverArray[@specialized A](val toArray: Array[A])(implicit tA: RType[A]
   @NeverInline
   def append(other: Coll[A]): Coll[A] = {
     if (toArray.length <= 0) return other
-    val result = CollectionUtil.concatArrays(toArray, other.toArray)
+    val result = if (VersionContext.current.isEvaluateErgoTreeUsingJIT) {
```

Comment:
explicit ErgoTree version comparison would be a better description here. Maybe because a comment is missed.

And as for other PRs not clear why versioning here is tree based while for other operations it is interpreter-based. Also, documentation is still missed.

Reply:
I am not enforcing. Both options are possible, but for tree-based versioning documentation should be done for application developer for possible versions, and how to choose a version. Here you pass a choice to a developer thus must document how to make it (otherwise, the work is simply not complete thus does not make sense).
Interpreter-based activation is implicit for a developer, so requires only for some list for interpreter developers, to track changes.

Reply:
Ok, changed to activation-based versioning. 
1) since this is semantic bug fix and, as you mentioned already, people usually expect the operation to have correct semantics
2) after activation it will simplify usability, as dealing with versions will not be required.

The tests are updated accordingly